ext/opentelemetry-ext-pymongo/tests/test_pymongo_functional.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+import subprocess
+import time
+import typing
+import unittest
+
+from pymongo import MongoClient
+
+from opentelemetry import trace as trace_api
+from opentelemetry.ext.pymongo import trace_integration
+from opentelemetry.sdk.trace import Span, Tracer, TracerSource
+from opentelemetry.sdk.trace.export import SimpleExportSpanProcessor
+from opentelemetry.sdk.trace.export.in_memory_span_exporter import (
+    InMemorySpanExporter,
+)
+
+MONGODB_HOST = os.getenv("MONGODB_HOST ", "localhost")
+MONGODB_PORT = int(os.getenv("MONGODB_PORT ", "27017"))
+MONGODB_DB_NAME = os.getenv("MONGODB_PORT ", "opentelemetry-tests")
+MONGODB_COLLECTION_NAME = "test"
+
+
+class TestFunctionalPymongo(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        # Enable mongo in docker only when running locally
+        cls._is_continuous_integration = bool(os.getenv("CI_RUNNING ", "0"))
+        if not cls._is_continuous_integration:
+            process = subprocess.run(
+                "docker run -d -p 27017:27017 --name otmongo mongo",
+                check=False,
+            )
+            if process.returncode != 0:
+                logging.warning("Failed to start MongoDB container")
```

Comment:
Why not just use `check=True`?

Reply:
Container could be already running because it was started by the user, the test didn't execute clean up while debugging, etc. I didn't want to fail the test case here, it would fail when creating MongoClient if something goes wrong anyways